### ISSUE TITLE
Is that possiable to get the decode output from ctc which repeated and blank symbols are not filtered 

### ISSUE THREAD

**Issue by mengya22** [SeanNaren/deepspeech.pytorch#136]
I want to get the ctc decode output sequence contained repeated and blank symbols and their probability of each symbols. Is that possiable to do in this project? Thanks a lot!

**Comment by ryanleary** [comment]
Yes, just implement your own decoder. The raw output of the network is essentially what you're looking for.

**Comment by mengya22** [comment]
@ryanleary  Thank you for your quick reply, you mean that I must add my own decode code so that I can get the output. Does the raw output contain blank symbols?

**Comment by mengya22** [comment]
@ryanleary Thank you for your quick reply, you mean that I must add my own decode code so that I can get the output. Can the CTCBeamDecoder function return raw output contain blank symbols? If so ,I can find the  probability  from the output of softmax.

**Comment by SeanNaren** [comment]
@mengya22 the CTCBeamDecoder will require you to modify the [pytorch-ctc](https://github.com/ryanleary/pytorch-ctc/blob/master/pytorch_ctc/__init__.py#L37) such that it outputs blanks and etc.

**Comment by mengya22** [comment]
@SeanNaren  Thanks a lot ! Does the warp-ctc implement a softmax layer inside the loss fuction? Or I should do the softmax layer by myself and put the output of softmax to the warp_ctc_loss fuction.

**Comment by SeanNaren** [comment]
@mengya22 warp-ctc does the softmax in the function, which is why we have this inference based softmax added to the network!

**Comment by mengya22** [comment]
@SeanNaren  I think I confuse the softmax layer and softmax regression. So in the CTCloss function,there are no train values in the softmax layer. And I can get the lables probability at each time step  by choose the max value index of rnn output, Is that correct? I am really new in this domian and I am so  sorry to  bother you.

**Comment by ryanleary** [comment]
Yes that is correct.
